Use `.interfaceimpl type` syntax

ltrzesniewski opened the pull request:
### Problem

I noticed the following output when decompiling an assembly:

![image](https://user-images.githubusercontent.com/7913492/219972190-f68753f1-7783-41fc-8313-00ec60ac2f79.png)

That `implements .custom` part seemed wrong, and indeed, `ildasm` uses another syntax instead: `.interfaceimpl type` ([see here](https://github.com/dotnet/runtime/blob/7908e8ef62c7d5ebdcd6459faccd755b9b10d369/src/coreclr/ildasm/dasm.cpp#L4847-L4870)).

Here is the result:

![image](https://user-images.githubusercontent.com/7913492/219973525-81877e20-069a-49d5-adb7-5985d16ec45f.png)

### Solution
* Any comments on the approach taken, its consistency with surrounding code, etc.

  This PR moves the custom attributes on interface implementations inside the type body and adds one `.interfaceimpl type` declaration for each interface implementation which has a custom attribute.

* Which part of this PR is most in need of attention/improvement?

  While implementing this, I noticed that `ilasm` is inconsistent with `ildasm` when multiple attributes are applied to an interface implementation. I believe `ilasm` is wrong here, and reported the following issue:  
  https://github.com/dotnet/runtime/issues/82373

  Because of this, the unit test only applies a single attribute on the interface implementation. I didn't want to add an additional `.expected.il` file just because of this inconsistency.

  This PR implements the same behavior as in `ildasm`.

  As for the UI, I wasn't sure if I should highlight the `type` keyword in green (which looked wrong) or in blue (which could lead to many false positives), so I left it as-is.

* [x] At least one test covering the code changed

ltrzesniewski commented:
BTW this is how dotPeek displays an interface implementation with two custom attributes:

![image](https://user-images.githubusercontent.com/7913492/219975738-29cb87bf-b45b-400a-9f53-a5274e6a61d6.png)

siegfriedpammer commented:
> * As for the UI, I wasn't sure if I should highlight the `type` keyword in green (which looked wrong) or in blue (which could lead to many false positives), so I left it as-is.

As `ilasm` treats `type` as a keyword and fails to assemble a method that is named `type`:  `error : syntax error at token 'type'`. I would say that you can safely highlight it, there would be no confusion in existing code.

ltrzesniewski commented:
Right, I should have tested it. I added the `type` keyword.

But upon checking, I noticed that single-quoted identifiers are still highlighted as kewords. This also applies to other keywords such as `instance`:

![image](https://user-images.githubusercontent.com/7913492/220070610-888dd5c5-89ec-4c15-9ef2-4a7749176440.png)

So I also fixed that in this PR:

![image](https://user-images.githubusercontent.com/7913492/220071249-209586f2-bece-4891-9e69-b01c1e34434c.png)
